### DISCUSSION TITLE
Do we need asynchronous commands execution?

### DISCUSSION THREAD

**Discussion by ndr-brt** [eclipse-edc/Connector#2672] [General]
The current commands implementation relies on an asynchronous execution using the `CommandQueue`.
This would require a persistent queue for reliability, otherwise the system could lost the commands, that currently are only stored in memory.
I'm not sure what are the advantages of having them asynchronous, I would propose to have them synchronous to avoid unnecessary complexity.

**Comment by jimmarino** [comment]
I think we should not go down the path of reliable async messaging. The async command queue attempted to address the issue of potential concurrent updates. I'd be in favor of making the command framework synchronous.
